src/adapter/z-stack/adapter/zStackAdapter.ts

```diff
@@ -217,7 +217,7 @@ class ZStackAdapter extends Adapter {
     ): Promise<Events.ZclDataPayload> {
         return this.queue.execute<Events.ZclDataPayload>(async () => {
             const command = zclFrame.getCommand();
-            if (!command.hasOwnProperty('response')) {
+            if (!command.hasOwnProperty('response') && zclFrame.Header.frameControl.disableDefaultResponse) {
```

Comment:
I don't understand this, a default response is something different than a response to a command, I don't think this is right.

Reply:
In case of error (example read/write to not supported cluster/attribute,...), the device responds with a defaultResponse.
See third example here: https://www.digi.com/support/forum/2086/default-response-for-zcl (second post)

The current matcher does not hit, because the commandId and direction of B->A defaultResponse is different than the read/write frame A->B was. So in current implementation, the NON-SUCCESS-defaultResonse can not be matched to a pending transaction.

Reply:
When a device doesn't support a certain attribute, it should respond with a write attributes response. Below is a screenshot when I'm trying to write `genBinaryInput` `activeText` attribute to a TRADFRI 5 button dimmer.

![image](https://user-images.githubusercontent.com/2892853/71640897-15749d00-2c93-11ea-96a4-2eb79c0907ee.png)

Reply:
> When a device doesn't support a certain attribute, it should respond with a write attributes response

Correct. But now read or write an unsupported **cluster**:

![grafik](https://user-images.githubusercontent.com/22577004/71642426-321ccf00-2cab-11ea-93bd-d7b6fa3b360a.png)

Reply:
I still cannot reproduce this, with the following device:
```
{"id":8,"type":"Router","ieeeAddr":"0x0017880104c9cd33","nwkAddr":49209,"manufId":4107,"manufName":"Philips","powerSource":"Mains (single phase)","modelId":"LWB010","epList":[11,242],"endpoints":{"11":{"profId":49246,"epId":11,"devId":256,"inClusterList":[0,3,4,5,6,8,4096],"outClusterList":[25],"clusters":{"genBasic":{"attributes":{"modelId":"LWB010","manufacturerName":"Philips","powerSource":1,"zclVersion":1,"appVersion":2,"stackVersion":1,"hwVersion":1,"dateCode":"20181115","swBuildId":"1.46.13_r26312"}},"seMetering":{"attributes":{}}},"binds":[]},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[33],"outClusterList":[33],"clusters":{},"binds":[]}},"appVersion":2,"stackVersion":1,"hwVersion":1,"dateCode":"20181115","swBuildId":"1.46.13_r26312","zclVersion":1,"interviewCompleted":true,"meta":{},"lastSeen":1578592996991}
```

I read the following value from an unsupported cluster:

```js
const device = this.resolveEntity("0x0017880104c9cd33");
device.endpoint.read('msTemperatureMeasurement', ['measuredValue']);
```

As you can see, it doesn't support this cluster, `msTemperatureMeasurement` = `1026`.

And I get the following response:

![image](https://user-images.githubusercontent.com/2892853/72092887-8e1ecd80-3313-11ea-9df8-00bbd5b4e79d.png)

Reply:
Did some tests (disableDefaultResponse=true):
## readAttribute to not existing cluster


- PhilipsHue, Ikea, ZBT(MuellerLicht) respond with `ReadAttributeResponse (status: unsupAttribute)`

- Osram  responds with `DefaultResponse (status: failure)`

- Eurotronics responds with `DefaultResponse (status: unsupCluster)`

## readAttribute to existing cluster but not existing Attribute

- all respond with `readAttrResp` with status unsupAttr)

## writeAttribute to existing cluster but read-only Attribute

- all respond with `readAttrResp` with status readOnly)

### Summary
A response in case of error can be both, normal `(read/..)AttributeResponse` or `DefaultResponse`, even if disableDefaultResponse=true. Depends again on manufacturer..

Reply:
Thanks, I will run some local tests but it may take some time.

Reply:
Thanks. No problem. I'm also ok if you prefer to implement it in another way. PR is just a suggest. It's only important, that the error/status codes reach the application.

Reply:
What Osram device did you try with? I've tried the code of  https://github.com/Koenkk/zigbee-herdsman/pull/114#discussion_r364885878 with https://www.zigbee2mqtt.io/devices/AC0251100NJ_AC0251700NJ.html  but it responds the same as https://github.com/Koenkk/zigbee-herdsman/pull/114#discussion_r364885878

Reply:
This one
https://www.zigbee2mqtt.io/devices/AB3257001NJ.html

Reply:
So it seems that the amount of devices affected by this is limited, how big of a problem is this?

Reply:
In my opinion it is important to handles this kind of response as well. The mentioned Osram Plug for instance is a very popular device (cheap and widely available). And as of the digi.com support link above, it even seems to be the correct behavior, to reply with defaultResponse in case of error.

Another suggest:
The condition  `if (!command.hasOwnProperty('response')... throw new Error(Command .. has no response, cannot wait for response)` may be removed completely, since there can always be a response (namely in case of error), even for commands that normally do not have a response.

Then we only need to make sure that `waitressValidator` matches **any** response, that results from our sent request (same sequence number), no matter if readAttrResp or defaultResp (implemented in PR already)

Reply:
Yes I understand, I will buy this device so I can investigate further. thanks!